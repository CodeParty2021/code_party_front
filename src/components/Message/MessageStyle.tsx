import { ReactSVG } from "react-svg";
import styled, { css } from "styled-components";
import { GRAY_90, RED_50, TURQUOISE_50, WHITE } from "styles/colors";
import { FONT } from "styles/constants/constants";
import { FlexGap } from "styles/FlexGap/FlexGap";

export type MessageStyleProps = {
  color?: "blue" | "red";
};

const defaultStyle = css`
  display: flex;
  flex-direction: column;
  align-items: center;

  .message_container_main {
    display: flex;
    flex-direction: column;
    align-items: center;
    ${FlexGap({ gap: "24px", direction: "column" })}

    padding: 72px 32px 0 32px;

    width: 614px;
    height: 210px;
    border-radius: 48px;
    background: ${WHITE};
  }

  .message_title {
    font-family: ${FONT.F_851Gkktt};
    font-size: 32px;
    line-height: 25px;
  }

  .message_value {
    font-size: 24px;
    line-height: 150%;
    color: ${GRAY_90};
    text-align: center;
  }
`;

const blueStyle = css`
  .message_title {
    color: ${TURQUOISE_50};
  }
`;

const redStyle = css`
  .message_title {
    color: ${RED_50};
  }
`;

export const MessageStyle = styled.div<MessageStyleProps>`
  ${defaultStyle}

  ${({ color }) => color == "blue" && blueStyle}
  ${({ color }) => color == "red" && redStyle}
`;

MessageStyle.defaultProps = {
  color: "blue",
};

export const MessageTriangleStyle = styled(ReactSVG)`
  width: 40px;
  height: 16px;
  fill: ${WHITE};
`;
