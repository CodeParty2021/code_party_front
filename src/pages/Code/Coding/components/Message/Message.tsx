import React from "react";
import {
  MessageStyle,
  MessageStyleProps,
  MessageTriangleStyle,
} from "./MessageStyle";

type Props = MessageStyleProps & {
  title: string;
  value: string;
};

export const Message: React.FC<Props> = ({
  title,
  value,
  ...styleProps
}) => {
  return (
    <MessageStyle {...styleProps}>
      <div className="message_container_main">
        <span className="message_title">{title}</span>
        <span className="message_value">{value}</span>
      </div>
      <MessageTriangleStyle
        src="/img/message_triangle.svg"
        wrapper="svg"
      />
    </MessageStyle>
  );
};
