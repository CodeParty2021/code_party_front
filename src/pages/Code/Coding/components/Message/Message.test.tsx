import React from "react";
import { Message } from "./Message";
import "jest-styled-components";
import { render } from "enzyme";

describe("<Message />", () => {
  it("snapshot test", () => {
    const wrapper = render(<Message title="タイトル" value="メッセージ" />);

    expect(wrapper).toMatchSnapshot();
  });
});
