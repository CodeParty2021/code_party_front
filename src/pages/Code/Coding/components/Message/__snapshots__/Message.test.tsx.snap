// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> snapshot test 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .message_container_main {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 24px;
  padding: 72px 32px 0 32px;
  width: 614px;
  height: 210px;
  border-radius: 48px;
  background: #FFF;
}

.c0 .message_title {
  font-family: "851Gkktt";
  font-size: 32px;
  line-height: 25px;
}

.c0 .message_value {
  font-size: 24px;
  line-height: 150%;
  color: #2A2C33;
  text-align: center;
}

.c0 .message_title {
  color: #0DB3C6;
}

.c1 {
  width: 40px;
  height: 16px;
  fill: #FFF;
}

@supports not (gap:24px) {
  .c0 .message_container_main > *:not(:first-child) {
    margin-top: 24px;
  }
}

<div
  class="c0"
  color="blue"
>
  <div
    class="message_container_main"
  >
    <span
      class="message_title"
    >
      タイトル
    </span>
    <span
      class="message_value"
    >
      メッセージ
    </span>
  </div>
  <svg
    class="c1"
    xlink="http://www.w3.org/1999/xlink"
    xmlns="http://www.w3.org/2000/svg"
  />
</div>
`;
